### PR TITLE
Documentation for home and sigs apps

### PR DESCRIPTION
**Pull request by ClayMav**
New to docstrings, let me know what you want changed. Be gentle.

### CODE REVIEW

File: ACM_General/home/views.py [hunk 1]
```diff
@@ -8,19 +8,38 @@
 
 
 def index(request):
-    # Grabs the 3 nearest upcoming events that haven't expired yet
+    """
+        Grabs the next 3 events that aren't past expiry,
```
Comment (kevinschoonover) [inline ACM_General/home/views.py]: No indent. And it's not guaranteed that it is three its with relation to settings.MAX_HOME_FLIER_COUNT.

File: ACM_General/home/views.py [hunk 2]
```diff
@@ -8,19 +8,38 @@
...
+            request,
+            'home/index.html',
+            {"upcoming_events": events}
+        )
+    )
 
 
 def sponsors(request):
+    """
+        Handles a request to see the sponsors page.
```
Comment (kevinschoonover) [inline ACM_General/home/views.py]: No Indent

File: ACM_General/home/views.py [hunk 3]
```diff
@@ -8,19 +8,38 @@
...
 
 def index(request):
-    # Grabs the 3 nearest upcoming events that haven't expired yet
+    """
+        Grabs the next 3 events that aren't past expiry,
+        then redirect to the index.
+
+        :param request: Request for index page with upcoming events.
+        :type request: Request
+        :rtype: Response
```
Comment (kevinschoonover) [inline ACM_General/home/views.py]: django.shortcut.render

File: ACM_General/home/views.py [hunk 4]
```diff
@@ -8,19 +8,38 @@
 
 
 def index(request):
-    # Grabs the 3 nearest upcoming events that haven't expired yet
+    """
+        Grabs the next 3 events that aren't past expiry,
+        then redirect to the index.
+
+        :param request: Request for index page with upcoming events.
+        :type request: Request
```
Comment (kevinschoonover) [inline ACM_General/home/views.py]: django.http.request.HttpRequest

File: ACM_General/home/views.py [hunk 5]
```diff
@@ -8,19 +8,38 @@
...
+        )
+    )
 
 
 def sponsors(request):
+    """
+        Handles a request to see the sponsors page.
+
+        :param request: Request for the sponsors page.
+        :type request: Request
```
Comment (kevinschoonover) [inline ACM_General/home/views.py]: See previous comment about type

File: ACM_General/home/views.py [hunk 6]
```diff
@@ -30,6 +49,14 @@ def sponsors(request):
 
 
 def calendar(request):
+    """
+        Handles a request to see the calendar page.
+
+        :param request: Request for the calendar page.
+        :type request: Request
```
Comment (kevinschoonover) [inline ACM_General/home/views.py]: See previous comment about type

File: ACM_General/home/views.py [hunk 7]
```diff
@@ -39,6 +66,14 @@ def calendar(request):
 
 
 def media(request):
+    """
+        Handles a request to see the media page.
+
+        :param request: Request for the media page.
+        :type request: Request
```
Comment (kevinschoonover) [inline ACM_General/home/views.py]: See previous comment about type

File: ACM_General/home/views.py [hunk 8]
```diff
@@ -48,6 +83,14 @@ def media(request):
 
 
 def officers(request):
+    """
+        Handles a request to see the officers page.
+
+        :param request: Request for the officers page.
+        :type request: Request
```
Comment (kevinschoonover) [inline ACM_General/home/views.py]: See previous comment about type

File: ACM_General/home/views.py [hunk 9]
```diff
@@ -57,6 +100,14 @@ def officers(request):
 
 
 def membership(request):
+    """
+        Handles a request to see the membership page.
+
+        :param request: Request for the membership page.
+        :type request: Request
```
Comment (kevinschoonover) [inline ACM_General/home/views.py]: See previous comment about type

File: ACM_General/sigs/managers.py [hunk 10]
```diff
@@ -4,20 +4,23 @@
...
     use_in_migrations = True
 
     def get_by_natural_key(self, id):
         """
-        TODO: Docstring
+        Gets the SIG by it's id
+
+        :param id: The id of the SIG 
+        :type id: int
+        :rtype: None
```
Comment (kevinschoonover) [inline ACM_General/sigs/managers.py]: sigs.models.SIG

File: ACM_General/sigs/managers.py [hunk 11]
```diff
@@ -4,20 +4,23 @@
...
+        :return: None
```
Comment (kevinschoonover) [inline ACM_General/sigs/managers.py]: Returns the a SIG model object with the id equal to the input specified id

File: ACM_General/sigs/managers.py [hunk 12]
```diff
@@ -4,20 +4,23 @@
...
         """
         return self.get(id=id)
 
     def _create_sig(self, **kwargs):
-        """
-        TODO: Docstring
-        """
+
```
Comment (kevinschoonover) [inline ACM_General/sigs/managers.py]: Add docstring

File: ACM_General/sigs/models.py [hunk 13]
```diff
@@ -56,6 +59,12 @@ class SIG(models.Model):
     )
 
     def __str__(self):
+        """
+        Returns the id of the SIG
+
+        :rtype: models.CharField
```
Comment (kevinschoonover) [inline ACM_General/sigs/models.py]: django.db.models.CharField

File: ACM_General/sigs/models.py [hunk 14]
```diff
@@ -56,6 +59,12 @@ class SIG(models.Model):
...
+        :return: self.id
```
Comment (kevinschoonover) [inline ACM_General/sigs/models.py]: The ID of the SIG.

File: ACM_General/home/views.py [hunk 15]
```diff
@@ -8,19 +8,38 @@
 
 
 def index(request):
-    # Grabs the 3 nearest upcoming events that haven't expired yet
+    """
+        Grabs the next 3 events that aren't past expiry,
+        then redirect to the index.
```
Comment (kevinschoonover) [inline ACM_General/home/views.py]: then renders the the template for the index page

File: ACM_General/home/views.py [hunk 16]
```diff
@@ -8,19 +8,38 @@
...
+
+        :param request: Request for index page with upcoming events.
```
Comment (kevinschoonover) [inline ACM_General/home/views.py]: Request object that contains information from the user's POST/GET request.

File: ACM_General/home/views.py [hunk 17]
```diff
@@ -8,19 +8,38 @@
...
+        :type request: Request
+        :rtype: Response
+        :return: Redirect to the base url with events, '/'.
```
Comment (kevinschoonover) [inline ACM_General/home/views.py]: The render template of the index page

File: ACM_General/home/views.py [hunk 18]
```diff
@@ -8,19 +8,38 @@
...
+            {"upcoming_events": events}
+        )
+    )
 
 
 def sponsors(request):
+    """
+        Handles a request to see the sponsors page.
+
+        :param request: Request for the sponsors page.
```
Comment (kevinschoonover) [inline ACM_General/home/views.py]: Request object that contains information from the user's POST/GET request.

File: ACM_General/home/views.py [hunk 19]
```diff
@@ -8,19 +8,38 @@
...
+        :type request: Request
+        :rtype: Response
+        :return: Redirect to the sponsors page, '/'.
```
Comment (kevinschoonover) [inline ACM_General/home/views.py]: The rendered template for the sponsors page

File: ACM_General/home/views.py [hunk 20]
```diff
@@ -39,6 +66,14 @@ def calendar(request):
 
 
 def media(request):
+    """
+        Handles a request to see the media page.
+
+        :param request: Request for the media page.
```
Comment (kevinschoonover) [inline ACM_General/home/views.py]: Request object that contains information from the user's POST/GET request.

File: ACM_General/home/views.py [hunk 21]
```diff
@@ -57,6 +100,14 @@ def officers(request):
 
 
 def membership(request):
+    """
+        Handles a request to see the membership page.
+
+        :param request: Request for the membership page.
+        :type request: Request
+        :rtype: Response
+        :return: Redirect to the membership page, '/'.
```
Comment (kevinschoonover) [inline ACM_General/home/views.py]: Rendered template for the membership page

File: ACM_General/sigs/managers.py [hunk 22]
```diff
@@ -4,20 +4,23 @@
...
     """
-    TODO: Docstring
+    This class contains functions to test all functions of SIG's
     """
     use_in_migrations = True
 
     def get_by_natural_key(self, id):
         """
-        TODO: Docstring
+        Gets the SIG by it's id
```
Comment (kevinschoonover) [inline ACM_General/sigs/managers.py]: Get the SIG model by its id

File: ACM_General/sigs/managers.py [hunk 23]
```diff
@@ -31,8 +34,22 @@ def _create_sig(self, **kwargs):
         return SIG
 
     def create_sig(self, **kwargs):
+        r"""
+        Creates a SIG
```
Comment (kevinschoonover) [inline ACM_General/sigs/managers.py]: Creates a SIG.

File: ACM_General/sigs/managers.py [hunk 24]
```diff
@@ -31,8 +34,22 @@ def _create_sig(self, **kwargs):
...
+        
+        :param \**kwargs: See below
```
Comment (kevinschoonover) [inline ACM_General/sigs/managers.py]: See Below.

File: ACM_General/sigs/tests.py [hunk 25]
```diff
@@ -62,6 +87,13 @@ def test_sig_model_member_functions(self):
         self.assertEqual(str(sig), sig.id)
 
     def test_sig_unique_constraint(self):
+        """
+        Verifies that you will get an error is trying to make two of the
```
Comment (kevinschoonover) [inline ACM_General/sigs/tests.py:91]: Verifies the uniqueness constraint for SIG creation (try to avoid using personal pronouns)

**Review by kevinschoonover — CHANGES_REQUESTED** [review]
Mostly the same error with Request and Response types as well as the return types for the functions. Make sure to put periods after the docstring description, return, and :params: tags. Also, don't indent the docstrings.

File: ACM_General/sigs/managers.py (hunk 12 above)
Comment (ClayMav) [reply ACM_General/sigs/managers.py]: I did, but the docstring does not render in Sphinx.

Comment (kevinschoonover) [reply ACM_General/sigs/managers.py]: It wont until Zach's `fix/sphinx-docs` branch is merged

**Pull request merged**